exporter/opentelemetry-exporter-otlp-json-common/src/opentelemetry/exporter/otlp/json/common/_internal/_log_encoder/__init__.py

```diff
@@ -0,0 +1,276 @@
+    if scope is None:
+        return ""
+    return f"{scope.name}|{scope.version}"
+
+
+def _encode_resource(resource: Resource) -> Dict[str, Any]:
+    """Encodes a resource into OTLP JSON format."""
+    if not resource:
+        return {"attributes": []}
+
+    return {
+        "attributes": _encode_attributes(resource.attributes),
+        "droppedAttributesCount": 0,  # Not tracking dropped attributes yet
+    }
+
+
+def _encode_instrumentation_scope(
+    scope: Optional[InstrumentationScope],
+) -> Dict[str, Any]:
+    """Encodes an instrumentation scope into OTLP JSON format."""
+    if scope is None:
+        return {"name": "", "version": ""}
+
+    return {
+        "name": scope.name or "",
+        "version": scope.version or "",
+        "attributes": [],  # Not using attributes for scope yet
+        "droppedAttributesCount": 0,
+    }
+
+
+def _encode_log_record(
+        log_data: LogData,
+        id_encoding: IdEncoding) -> Dict[str, Any]:
+    """Encodes a log record into OTLP JSON format."""
+    log_record = log_data.log_record
+
+    result = {
+        "timeUnixNano": str(log_record.timestamp),
+        "observedTimeUnixNano": str(
+            getattr(log_record, "observed_timestamp", log_record.timestamp)
+        ),
+        "severityNumber": _get_severity_number_value(
+            log_record.severity_number
+        ),
+        "severityText": log_record.severity_text or "",
+        "attributes": _encode_attributes(log_record.attributes),
+        "droppedAttributesCount": getattr(log_record, "dropped_attributes", 0),
+    }
+
+    # Handle body based on type
+    if log_record.body is not None:
+        result.update(_encode_any_value(log_record.body))
+
+    # Handle trace context if present
+    if log_record.trace_id:
+        result["traceId"] = encode_id(id_encoding, log_record.trace_id, 16)
+
+    if log_record.span_id:
+        result["spanId"] = encode_id(id_encoding, log_record.span_id, 8)
+
+    if (
+        hasattr(log_record, "trace_flags")
+        and log_record.trace_flags is not None
+    ):
+        result["flags"] = int(log_record.trace_flags)
+
+    return result
+
+
+def _encode_attributes(attributes: Dict[str, Any]) -> List[Dict[str, Any]]:
+    """Encodes attributes into OTLP JSON format."""
+    if not attributes:
+        return []
+
+    attribute_list = []
+    for key, value in attributes.items():
+        if value is None:
+            continue
+
+        attribute = {"key": key}
+        attribute.update(_encode_attribute_value(value))
+        attribute_list.append(attribute)
+
+    return attribute_list
+
+
+# pylint: disable=too-many-return-statements
+def _encode_attribute_value(value: Any) -> Dict[str, Any]:
+    """Encodes a single attribute value into OTLP JSON format."""
+    if isinstance(value, bool):
+        return {"value": {"boolValue": value}}
+    if isinstance(value, int):
+        return {"value": {"intValue": value}}
```

Comment:
Sadly that's incorrect.

Recall that JavaScript coerces large integers to floating point.

I believe this is spelled out in the OTLP spec.